lib/livereload.js

```diff
@@ -6,126 +6,116 @@
+        }
+    });
 
-    var defaultOptions = {
-        server: {
-            baseDir: self.projectRoot,
-            directory: true
-        },
-        open: false,
+    // If none of the supplied platforms are supported, stop and return an error
+    if (platforms === undefined || platforms.length === 0) {
+        var msg = 'None of the platforms supplied are currently supported for LiveReload.';
+        livereloadEvents.emit('livereload:error', msg);
+        return Q.reject(msg);
+    }
+
+    var BrowserSyncServer = require('./browserSyncServer');
+        
+    // If user has entered whitespaces surrounding '--ignore', 
+    // ... we end up having helpers.parseOptions() return 'ignore' with value true instead of a string,
+    // ... which can't be used as a string. So, Let's check for those cases
+    // e.g of error cases: 
+    //      `cordova run android -- --livereload --ignore, // with no path to ignore
+    //      `cordova run android -- --livereload --ignore= css, // with space after the '=' sign
+    //      etc...
+    // ToDO: ignore whitespaces surrounding the --ignore option or at least display a warning message to user
+    var ignoreOption = undefined;
+    if (options.ignore) {
+        ignoreOption = path.join(projectRoot, 'www', options.ignore);
+    }
+ 
+    // ToDO: What if user specifies custom option that clutters watchOptions ? tunnel ? files ? other options ?
+    var bs = new BrowserSyncServer(projectRoot, {
+        files: [{
+            match: [path.join(projectRoot, 'www', '**/*.*')],
+            fn: function (event, file) {
+                
+                // This is used by external clients (consumers of this npm package)
+                var externalAPI = {
+                    stop: bs.stopServer,
+                    reloadBrowsers: bs.reloadBrowsers,
+                    reloadFile: bs.reloadFile,
+                    browserSync: bs.browserSync
+                };
 
-        // ToDO: Document that if fn is overriden by another option, multi-device scenario 
-        // ... will be lost due to 'monkey patching' going away
-        snippetOptions: {
-            rule: {
-                match: /<\/body>/i,
-                fn: function(snippet, match) {
-                    return monkeyPatch() + snippet + match;
-                }
+                return Q().then(function () {
+                    options.cb(event, file, externalAPI);
+                    return Q();
+                }).fail(function (err) {
+                    livereloadEvents.emit('livereload:error', err, externalAPI);
+                });
             }
+        }],
+        watchOptions: {
+
+            // If user specified files/folders to ignore (via `cordova run android -- --livereload --ignore=build/**/*.*`), ignore those files
+            // ... Otherwise, don't ignore any files (That's the default). The function below achieves that goal.
+            ignored: ignoreOption || function (str) {
```

Comment:
Yes. Actually, this is exposed by chokidar, which BrowserSync uses: https://github.com/paulmillr/chokidar#path-filtering